src/Viewer.tsx

```diff
@@ -248,10 +248,12 @@ function Viewer(): ReactElement {
     scatterPlotConfig,
   ]);
 
-  // Update url whenever the viewer settings change
-  // (but not while playing/recording for performance reasons)
+  // Update url whenever the viewer settings change, with a few exceptions:
+  // - The URL should not change while playing/recording for performance reasons.
+  // - The URL should not change if the dataset hasn't loaded yet, or if it failed to load.
+  //   that way, users can refresh the page to try again.
   useEffect(() => {
-    if (!timeControls.isPlaying() && !isRecording) {
+    if (!timeControls.isPlaying() && !isRecording && isInitialDatasetLoaded) {
       setSearchParams(getUrlParams(), { replace: true });
     }
   }, [timeControls.isPlaying(), isRecording, getUrlParams]);
```

Comment:
should `isInitialDatasetLoaded` be a dependency?

Reply:
Yes, good catch!